File: ApiAgendamento/Domain/Entities/Medico.cs
namespace ApiAgendamento.Domain.Entities;

public class Medico
{
    public int Id { get; private set; }
    public string? Nome { get; private set; }
    public string? Especialidade { get; private set; }

    public List<HorarioDisponivel> HorariosDisponiveis { get; private set; } = new();

    protected Medico() { }

    public Medico(string nome, string especialidade)
    {
        Nome = nome;
        Especialidade = especialidade;
    }

}

File: frontend-app/src/models/Medico.ts
import type { HorarioDisponivel } from '@/models/HorarioDisponivel'
// Modelo (entidade) para Medico
export interface Medico {
  id: number;
  nome: string;
  especialidade: string;
}

export interface AgendaMedico {
  id: number;
  dataHora: string;
  pacienteId: number;
  pacienteNome: string;
  pacienteEmail: string;
}

export interface MedicoComHorarios {
  medicoId: number;
  nome: string;
  especialidade: string;
  horariosDisponiveis: HorarioDisponivel[];
}
